### ISSUE TITLE
vipsthumbnail with export profile srgb causing artifacts in output image

### ISSUE THREAD

**Issue by frederikrosenberg** [libvips/libvips#4301]
**Describe the bug**
When using `vipsthumbnail` with the flag `--export-profile srgb` on the provided tif image the output image contains artifacts. If the `--export-profile` is left out the output image contains no artifacts. 

**To Reproduce**
1. Use the image '13103.tif' found in the provided zip file
2. Use the following command on the image `vipsthumbnail 13103.tif -o "13103.webp[keep=none]" --size 1000x --export-profile srgb`. Size is not relevant, just to make the artifacts more obvious.
3. Obverse the artifacts on the output image
4. Remove `--export-profile srgb` from the command and run it again
5. Obverse no artifacts on the output image

**Expected behavior**
Expect no artifacts to be present

**Actual behavior**
Artifacts are present

**Environment**
 - OS: Ubuntu 22.04.5 LTS (wsl)
 - Vips: 8.16.0

**Additional context**
[13103.zip](https://github.com/user-attachments/files/18005032/13103.zip)

This bug has been found in an application using NetVips to process images in C# running version NetVips version 2.4.1 and vips 8.15.2.
I have been able to reproduce on the command line. More arguments are used in the real application but seems to be irrelevant for this bug. The bug also appears for other export types such as PNG, JPG.

**Comment by jcupitt** [comment]
Hi @frederikrosenberg,

Thanks for this report. It seems the case that's breaking is 16-bit premultiply / resize / unpremultiply. If I run:

```
$ vips copy 13103.tif t1.v
$ vips premultiply t1.v t2.v
$ vips resize t2.v t3.v 0.9
$ vips unpremultiply t3.v t4.png
```

You'll see classic ringing premultiply artifacts in the output, but if I convert to 8-bit first:

```
$ vips colourspace 13103.tif t1.v srgb
$ vips premultiply t1.v t2.v
$ vips resize t2.v t3.v 0.9
$ vips unpremultiply t3.v t4.png
```

It's all fine. I'm not sure why the 16-bit case is failing, I'll keep digging.

**Comment by jcupitt** [comment]
Wait, that's not right, there's a bug in vipsdisp :( I'll keep looking.

**Comment by jcupitt** [comment]
Ah, finally found it, it's an issue exporting 16-bit RGBA to 8-bit. Test 16-bit RGBA PNG:

![t3](https://github.com/user-attachments/assets/1d206130-5569-4355-94e1-a5558196f8ed)

Run:

```
vips icc_transform t3.png x.png srgb --depth 8
```

And you get:

![x](https://github.com/user-attachments/assets/a6148653-330e-40e2-934a-2206cf27d747)

**Comment by jcupitt** [comment]
I found a fix (a one line change, sigh) and made a PR. Let's move any discussion there.

Thanks again for reporting this.

**Comment by jcupitt** [comment]
... I credited you, I hope that's OK. This fix will be in 8.16.1.

As a workaround until this fix is widely available, you can convert to srgb before thumbnailing, eg. perhaps:

```C#
var thumb = image.colourspace("srgb").thumbnail(...)
```

(though I think net-vips will prefer an enum to a string here)

**Comment by frederikrosenberg** [comment]
Thanks for the work around and quick fix!

I'm fine with being credited, thanks again for an amazing library!
